src/manifold/src/csg_tree.h

```diff
@@ -22,13 +22,16 @@ enum class CsgNodeType { Union, Intersection, Difference, Leaf };
 
 class CsgLeafNode;
 
-class CsgNode {
+class CsgNode : public std::enable_shared_from_this<CsgNode> {
```

Comment:
Thanks! I never know such a thing exists, I wanted to do something similar and have to do it the dumb way.

Reply:
Yeah it's a nice get out of jail free card 🤣